File: Data/Journals/Journals.Data.Entity/Models/Address.cs
namespace ProcessingTools.Journals.Data.Entity.Models
{
    using System;
    using System.ComponentModel.DataAnnotations;
    using ProcessingTools.Journals.Data.Common.Constants;
    using ProcessingTools.Journals.Data.Common.Contracts.Models;

    public class Address : IAddress
    {
        public Address()
        {
            this.Id = Guid.NewGuid().ToString();
        }

        [Key]
        [Required(AllowEmptyStrings = false)]
        [MaxLength(ValidationConstants.MaximalLengthOfId)]
        public string Id { get; set; }

        [Required(AllowEmptyStrings = false)]
        [MaxLength(ValidationConstants.MaximalLengthOfAddressString)]
        public string AddressString { get; set; }

        public int? CityId { get; set; }

        public int? CountryId { get; set; }
    }
}

File: src/ProcessingTools/Web.Core/ProcessingTools.Web.Documents/ClientApp/code/js/temp/Areas/Journals/Models/Shared/Address.js
function Address(id, addressString, cityId, countryId, status) {
    var self = this;

    self.id = id;
    self.addressString = ko.observable(addressString).extend({ required: true, minLength: 3 });
    self.cityId = ko.observable(cityId).extend({ required: true });
    self.countryId = ko.observable(countryId).extend({ required: true });
    self.status = ko.observable(status);
}
